system/ee/legacy/libraries/Grid_parser.php

```diff
@@ -247,6 +247,10 @@ public function parse($channel_row, $field_id, $params, $tagdata, $content_type
 
         $columns = ee()->grid_model->get_columns_for_field($field_id, $content_type);
 
+        // since the get_columns_for_field was passed a single filed ID and NOT
+        // an array we need to make it an array as pre_loop expects an array
+        $this->_pre_loop(array($columns));
```

Comment:
If I am reading the code right, `get_columns_for_fields` returns an array - no matter what you passed it. So here you looks like you are making it array of arrays?

Also, what's the reason we need `_pre_loop` to be added here?